faker/providers/ssn/en_US/__init__.py

```diff
@@ -1,19 +1,172 @@
+            '05',
+            '06',
+            '10',
+            '11',
+            '12',
+            '13',
+            '14',
+            '15',
+            '16',
+            '20',
+            '21',
+            '22',
+            '23',
+            '24',
+            '25',
+            '26',
+            '27',
+            '30',
+            '31',
+            '32',
+            '33',
+            '34',
+            '35',
+            '36',
+            '37',
+            '38',
+            '39',
+            '40',
+            '41',
+            '42',
+            '43',
+            '44',
+            '45',
+            '46',
+            '47',
+            '48',
+            '50',
+            '51',
+            '52',
+            '53',
+            '54',
+            '55',
+            '56',
+            '57',
+            '58',
+            '59',
+            '60',
+            '61',
+            '62',
+            '63',
+            '64',
+            '65',
+            '66',
+            '67',
+            '68',
+            '71',
+            '72',
+            '73',
+            '74',
+            '75',
+            '76',
+            '77',
+            '80',
+            '81',
+            '82',
+            '83',
+            '84',
+            '85',
+            '86',
+            '87',
+            '88',
+            '90',
+            '91',
+            '92',
+            '93',
+            '94',
+            '95',
+            '98',
+            '99']
+
+        ein_prefix = random.choice(ein_prefix_choices)
+        sequence = self.random_int(min=0, max=9999999)
+
+        ssn = "{0:s}-{1:07d}".format(ein_prefix, sequence)
+        return(ssn)
+
+    def ssn(self, taxpayer_identification_number_type=SSN_TYPE):
+        """ Generate a random United States Taxpayer Identification Number of the specified type.
+
+        If no type is specified, a US SSN is returned.
+        """
+
+        if taxpayer_identification_number_type.lower() == self.ITIN_TYPE:
+            return self.itin()
+        elif taxpayer_identification_number_type.lower() == self.EIN_TYPE:
+            return self.ein()
+        elif taxpayer_identification_number_type.lower() == self.SSN_TYPE:
+
+            if taxpayer_identification_number_type.lower() == 'ssn':
+
+                # Certain numbers are invalid for United States Social Security
+                # Numbers. The area (first 3 digits) cannot be 666 or 900-999.
+                # The group number (middle digits) cannot be 00. The serial
+                # (last 4 digits) cannot be 0000.
+            
+                area = self.random_int(min=1, max=899)
+                if area == 666:
+                    area += 1
+                group = self.random_int(1, 99)
+                serial = self.random_int(1, 9999)
+
+                ssn = "{0:03d}-{1:02d}-{2:04d}".format(area, group, serial)
+                return(ssn)
+    
+        else:
+            raise ValueError
```

Comment:
can you add a descriptive message to the `ValueError`? something like `raise ValueError("`taxpayer_identification_number_type` must be 'ssn', 'ein' or '' itin'.")`